presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/QueryBuilder.java

```diff
@@ -0,0 +1,364 @@
+        }
+    }
+
+    private static boolean isAcceptedType(Type type)
+    {
+        Type validType = requireNonNull(type, "type is null");
+        return validType.equals(BigintType.BIGINT) ||
+                validType.equals(TinyintType.TINYINT) ||
+                validType.equals(SmallintType.SMALLINT) ||
+                validType.equals(IntegerType.INTEGER) ||
+                validType.equals(DoubleType.DOUBLE) ||
+                validType.equals(RealType.REAL) ||
+                validType.equals(BooleanType.BOOLEAN) ||
+                validType.equals(DateType.DATE) ||
+                validType.equals(TimeType.TIME) ||
+                validType.equals(TimeWithTimeZoneType.TIME_WITH_TIME_ZONE) ||
+                validType.equals(TimestampType.TIMESTAMP) ||
+                validType.equals(TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE) ||
+                validType instanceof VarcharType ||
+                validType instanceof CharType ||
+                validType instanceof DecimalType;
+    }
+
+    private static List<String> toConjuncts(List<PhoenixColumnHandle> columns, TupleDomain<ColumnHandle> tupleDomain, List<TypeAndValue> accumulator)
+    {
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+        for (PhoenixColumnHandle column : columns) {
+            Type type = column.getColumnType();
+            if (isAcceptedType(type)) {
+                Domain domain = tupleDomain.getDomains().get().get(column);
+                if (domain != null) {
+                    builder.add(toPredicate(column.getColumnName(), domain, type, accumulator));
+                }
+            }
+        }
+        return builder.build();
+    }
+
+    private static String toPredicate(String columnName, Domain domain, Type type, List<TypeAndValue> accumulator)
+    {
+        checkArgument(domain.getType().isOrderable(), "Domain type must be orderable");
+
+        if (domain.getValues().isNone()) {
+            return domain.isNullAllowed() ? columnName + " IS NULL" : "FALSE";
+        }
+
+        if (domain.getValues().isAll()) {
+            return domain.isNullAllowed() ? "TRUE" : columnName + " IS NOT NULL";
+        }
+
+        List<String> disjuncts = new ArrayList<>();
+        List<Object> singleValues = new ArrayList<>();
+        for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
+            checkState(!range.isAll()); // Already checked
+            if (range.isSingleValue()) {
+                singleValues.add(range.getLow().getValue());
+            }
+            else {
+                List<String> rangeConjuncts = new ArrayList<>();
+                if (!range.getLow().isLowerUnbounded()) {
+                    switch (range.getLow().getBound()) {
+                        case ABOVE:
+                            rangeConjuncts.add(toPredicate(columnName, ">", range.getLow().getValue(), type, accumulator));
```

Comment:
I would feel safer if we could reuse base-jdbc QueryBuilder instead of copying the class